red_queen/cli.py

```diff
@@ -0,0 +1,172 @@
+    help="enter a compliler here",
+)
+@click.option(
+    "-t",
+    "--benchmarkType",
+    "benchmarkType",
+    multiple=True,
+    type=click.Choice(benchmark_types),
+    help="enter the type of benchmark(s) here",
+)
+@click.option(
+    "-b",
+    "--benchmark",
+    "benchmark",
+    multiple=True,
+    type=click.Choice(benchmarks),
+    help="enter the specfic benchmark(s) here",
+)
+def main(compiler, benchmarkType, benchmark):
+    benchmarkPaths = []
+    pytestPaths = ""
+    myDict = {}
+    mTag = ""
+    if len(compiler) > 0:
+        mTag = "-m "
+        compiler = compiler[0]
+    else:
+        compiler = ""
+    # ### This for loop ensures that we are able to run various benchmark types
+    i = 0
+    j = 0
+    ### Has a benchmark type been specified?
+    if len(benchmarkType) > 0:
+        # click.echo("passed test 0")
+        while i < len(benchmarkType):
+            ### Is the inputted benchmark type valid?
+            if set(benchmarkType).issubset(benchmark_category):
+                # click.echo("passed test 1")
+                ### Has a benchmark been specified?
+                if len(benchmark) > 0:
+                    # click.echo("passed test 2")
+                    ### Are the inputted benchmark(s) valid?
+                    if set(benchmark).issubset(benchmarks):
+                        # click.echo("passed test 3")
+                        ### Is the inputted benchmark within the inputted benchmark type suite?
+                        if set(benchmark).issubset(set(benchmark_category[benchmarkType[i]])):
+                            # click.echo("passed test 4")
+                            for j in range(len(benchmark)):
+                                benchmarkPaths.append(
+                                    benchmark_category[benchmarkType[0]][benchmark[j]]
+                                )
+                            pytestPaths = " ".join(tuple(benchmarkPaths))
+                            runBenchmarks(pytestPaths, windows_ad1, mTag, compiler)
+                            showResult()
+                            i += 1
+                else:
+                    myDict = benchmark_category[benchmarkType[0]]
+                    for v in myDict.values():
+                        benchmarkPaths.append(v)
+                    pytestPaths = " ".join(tuple(benchmarkPaths))
+                    runBenchmarks(pytestPaths, windows_ad1, mTag, compiler)
+                    showResult()
+                    i += 1
+    else:
+        question = input(f"Would you like to run all {len(benchmarks)} available benchmarks (y/n) ")
+        if question.lower() == "y":
+            for benchmark_list in benchmark_category.values():
+                for paths in benchmark_list.values():
+                    benchmarkPaths.append(paths)
+            pytestPaths = " ".join(tuple(benchmarkPaths))
+            runBenchmarks(pytestPaths, windows_ad1, mTag, compiler)
+            showResult()
```

Comment:
This might be a personal thing, but maybe showing the result should be an option instead of automatically doing it. like it could be an argument in the cli like:
```runBenchmarks(pytestPaths, windows_ad1, mTag, compiler, results)```
and then:
```
def runBenchmarks(pytestPaths: str, windows_ad1: str, mTag: str, compiler: str, results: str):
     click.echo("benchmarks ran")
     subprocess.run(
         [f"{windows_ad1}pytest -n auto {windows_ad2}{pytestPaths} {mTag}{compiler} --store"],
         shell=True,
     )
     if results == "table":
         showResults()
```
or something like that, just a suggestion